walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.Comparator;
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Arrays;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+
+    private final ArrayList < TimeRange > options = new ArrayList < TimeRange > ();
+
+    public Collection < TimeRange > query(Collection < Event > events, MeetingRequest request) {
+
+        Collection attendees = request.getAttendees();
+        Collection optionalAttendees = request.getOptionalAttendees();
+
+        // check if duration is longer than the day. if so, return empty collection.
+        long duration = request.getDuration();
+        if (duration > TimeRange.WHOLE_DAY.duration()) {
+            return Arrays.asList();
+        }
+
+        // if no attendees or optional attendees return collection of one item being the full day
+        if (attendees.isEmpty() && optionalAttendees.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+
+        // if there are optional attendees but no attendees, optional attendees will be treated as attendees
+        } else if (attendees.isEmpty()) {
+            attendees = optionalAttendees;
+        }
+
+        // if no events return the whole day
+        ArrayList < Event > eventsList = new ArrayList < Event > (events);
+        if (eventsList.isEmpty() == true) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+
+        // Sorts array of event objects by start time in ascending order
+        Collections.sort(eventsList, Event.ORDER_BY_START_TIME);
+        Iterator < Event > eventsListIterator = eventsList.iterator();
+
+        // Conflicting event is found
+        Event conflict = eventsListIterator.next();
+        int start;
+
+        // Ignore conflict if it doesn't involve any person in the request 
+        if (conflictIsRelevant(conflict, attendees) == false) {
```

Comment:
a nit re method naming. naming is a dark art full of contradicting rules, but generally if there's verb in the name you want to consider having it come first. This method would be better named "isConflictRelevant" or even "isConflict". This will come more naturally to you as you read/write more code.

Reply:
done